src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientMetrics.java

```diff
@@ -29,10 +29,17 @@
     String CACHE_MISS_METRIC = "graphql-client.cache.misses";
     String CACHE_EVICTION_METRIC = "graphql-client.cache.evictions";
     String CACHE_USAGE_METRIC = "graphql-client.cache.usage";
+    String CONNECTION_POOL_AVAILABLE_METRIC = "graphql-client.connection-pool.available-connections";
+    String CONNECTION_POOL_PENDING_METRIC = "graphql-client.connection-pool.pending-requests";
+    String CONNECTION_POOL_USAGE_METRIC = "graphql-client.connection-pool.usage";
 
     GraphqlClientMetrics NOOP = new GraphqlClientMetrics() {
 
-        @Override public void addCacheMetric(String metric, String cacheName, Supplier<? extends Number> valueSupplier) {
+        @Override public void addConnectionPoolMetric(String metricName, Supplier<? extends Number> valueSupplier) {
```

Comment:
Instead of adding more and more methods to the NOOP instance, you could try using a [Proxy](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/Proxy.html). Less code and less code to cover by unit tests.

Reply:
I would prefer to keep this as is, without introducing reflection API usage.